lib/browser-api.js

```diff
@@ -592,7 +592,16 @@ module.exports = function (browserWindow, panel, webview) {
       'windowShouldClose:': function (
         notification
       ) {
-        this.utils.emit('close') // TODO: prevent default
+        var shouldClose = true
+        this.utils.emit('close', {
+          get defaultPrevented () {
+            return !shouldClose
+          },
+          preventDefault: function () {
+            shouldClose = false
+          }
+        })
+        return shouldClose
```

Comment:
I don't quite understand where this gets checked, but `preventDefault` doesn't seem to have an impact.

With this handler the window still closes.

```JavaScript
this.ui.on('close', event => {
  event.preventDefault();
});
```

I tried editing this snippet directly to set the value of `shouldClose` manually and it didn't have any impact on whether the window closes.

Reply:
it's based on https://developer.apple.com/documentation/appkit/nswindowdelegate/1419380-windowshouldclose?language=objc